modules/swagger-codegen/src/main/resources/JavaSpring/apiController.mustache

```diff
@@ -54,35 +52,36 @@ public class {{classname}}Controller implements {{classname}} {
 
 {{/isDelegate}}
 {{^jdk8-no-delegate}}
-{{#operation}}
-    public {{#async}}Callable<{{/async}}ResponseEntity<{{>returnTypes}}>{{#async}}>{{/async}} {{operationId}}({{#allParams}}{{>queryParams}}{{>pathParams}}{{>headerParams}}{{>bodyParams}}{{>formParams}},
-        {{/allParams}}@RequestHeader(value = "Accept", required = false) String accept) throws Exception {
-        // do some magic!
-{{#useSpringCloudClient}}
-        {{^isDelegate}}
+    {{^isDelegate}}
         {{^async}}
-        return new ResponseEntity<{{>returnTypes}}>(HttpStatus.OK);
-        {{/async}}
-        {{#async}}
-        return new Callable<ResponseEntity<{{>returnTypes}}>>() {
-            @Override
-            public ResponseEntity<{{>returnTypes}}> call() throws Exception {
-                return new ResponseEntity<{{>returnTypes}}>(HttpStatus.OK);
-            }
-        };
+    private static final Logger log = LoggerFactory.getLogger({{classname}}Controller.class);
+
+    private final ObjectMapper objectMapper;
+
+    @org.springframework.beans.factory.annotation.Autowired
+    private HttpServletRequest request;
```

Comment:
Why isn't it `((ServletRequestAttributes) RequestContextHolder .getRequestAttributes()).getRequest();` as you proposed previously? Consider using constructor injection (for consistency at least, see ObjectMapper).

Reply:
I didn't see your comment. Yes constructor injection is better. I was just not sure initially that it worked with scoped proxy beans but apparently it does.